go-controller/pkg/ovn/namespace.go

```diff
@@ -245,7 +245,7 @@ func (oc *Controller) updateNamespace(old, newer *kapi.Namespace) {
 	if gwAnnotation != oldGWAnnotation || newBFDEnabled != oldBFDEnabled {
 		// if old gw annotation was empty, new one must not be empty, so we should remove any per pod SNAT
 		if oldGWAnnotation == "" {
-			if config.Gateway.DisableSNATMultipleGWs && (len(nsInfo.routingExternalGWs.gws) != 0 || len(nsInfo.routingExternalPodGWs) != 0) {
```

Comment:
change seems logical, but you have to consider the external pod gw path I think. If a pod is added to serve as an external gateway, we also need to remove the SNAT then.

Reply:
Yes, that is what #2272 fixes.